### ISSUE TITLE
Rule-Based Styled layer won't show in leaflet

### ISSUE THREAD

**Issue by markseb44** [tomchadwin/qgis2web#620]
Hi, using QGIS 3. Im brand new to this but it looks so close to being an excellent tool for me.

Have 3 layers only. Did some testing and all was fine until I used Rule-Based styling on 1 layer. Then when I published using leaflet I got a blank page. I researched and tried various  fixes... Putting the styled layer on top allowed the other 2 layers 2 show. I removed highlighting. I removed the show layers button. I have tried various browsers and have removed all layers except the styled layer.

All to no avail. Sorry I am no programmer, but do a lot of GIS. 

Any help really appreciated!!
Attached zip file has all the files in it. 

[css.zip](https://github.com/tomchadwin/qgis2web/files/1765542/css.zip)

**Comment by tomchadwin** [comment]
I think the issue is that the combination of a pattern fill with a rule-based style is failing. Definitely looks like a bug, and seems to affect categorized styles, too. Can you change to a simple fill, and confirm whether the map comes back to life? Meanwhile, I'll investigate why this combination might fail.

**Comment by tomchadwin** [comment]
OK, that took more fixing than I anticipated - apologies for closing the issue early. Can you see if there is any improvement using master branch and report back?

**Comment by markseb44** [comment]
my apologies...i don't know how to do that :(

Sorry, my programming skills go as far as running an executable to install
the program!....

**Comment by tomchadwin** [comment]
Here's how to do it:

1. Go to https://github.com/tomchadwin/qgis2web/
2. Click the green "Clone or download" button
3. Choose the zip option
4. Open the downloaded zipfile, and open the folder within it called `qgis2web-master`
5. Open the subfolder `qgis2web`
6. Copy all the files and folders in there
7. Find the qgis2web installation folder on your machine (where it is depends on whether you are on Windows, Mac, etc, so if you don't know where to find it, just reply with what OS you use)
8. Paste the files you copied from the zip into the qgis2web folder, overwriting what's there
9. Restart QGIS

That will give you the changes I made in response to your issue, so if you are able to do so, and report back on how it works (or not), that would be incredibly helpful.

**Comment by tomchadwin** [comment]
For step 7 above, to find the qgis2web installation folder, do the following:

1. In QGIS, click `Settings > User Profiles > Open active profile folder`
2. In the folder which opens, drill down to `python/plugins/qgis2web`

Alternatively, you can create a whole new user profile for this testing work, so you can keep your main QGIS user profile on released versions of plugins, rather than Github downloads.

**Comment by markseb44** [comment]
Hi,

And thanks for your help.  I did as you asked. I can now see all the
layers, including my paddock layer which was missing. But the rule-based
coloring is not working. All paddocks are hatched red, with no
differentiation. The other thing is, when I mouse over them they change
color to yellow [highlight color], and then they change color (to blue)
when I mouse away?

I tried this on Chrome and IE, with same results.

Cheers
Mark

**Comment by tomchadwin** [comment]
Can you strip all other layers out from your project, zip together your project and layer files, and upload here? I can then try to recreate the issue you are having.

**Comment by tomchadwin** [comment]
Closing with no feedback - comment again and I can reopen.

**Comment by MonikaBarget** [comment]
I only recently installed QGIS2web and have the same problem. I used a GeoJSON file with about 350 individual geometries (all points) to create different sub-layers based on rules. Exporting simple layers in QGIS2web worked great for me. No I got the blank screen described by the other user. So I decided to duplicate my layer and select only one sub-set each (cf. screenshot) I could at least get QGIS2web to export this correctly: the layers are in the "layers" folder in .js format and the content is correct. But they are still not displayed when I open the HTML file. This time, I see an empty base map.

![grafik](https://user-images.githubusercontent.com/38257338/81787605-970dee80-9501-11ea-8733-d08d88843b8e.png)

I can put the relevant files in one of my own GIT repositories if you are still willing to re-open the issue. Thank you very much.
